x-pack/solutions/security/plugins/security_solution/common/detection_engine/rule_management/to_simple_rule_schedule.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { calcDateMathDiff } from '@kbn/securitysolution-utils/date_math';
+import { toLargestTimeDuration } from '@kbn/securitysolution-utils/time_duration';
+import type { RuleSchedule } from '../../api/detection_engine';
+import type { SimpleRuleSchedule } from '../../api/detection_engine/model/rule_schema/simple_rule_schedule';
+
+/**
+ * Transforms RuleSchedule to SimpleRuleSchedule by replacing `from` and `to` with `lookback`.
+ *
+ * The transformation is only possible when `to` equals to `now` and result `lookback` is non-negative.
+ */
+export function toSimpleRuleSchedule(ruleSchedule: RuleSchedule): SimpleRuleSchedule | undefined {
+  if (ruleSchedule.to !== 'now') {
+    return;
+  }
+
+  const lookBackMs = calcDateMathDiff(ruleSchedule.from, `now-${ruleSchedule.interval}`);
+
+  if (lookBackMs === undefined || lookBackMs < 0) {
+    return;
+  }
+
+  return {
+    interval: ruleSchedule.interval,
+    lookback: toLargestTimeDuration(lookBackMs),
+  };
+}
```

Comment:
The fact that we convert it `toLargestTimeDuration` - could it cause rules to be incorrectly marked as customized? Or we are supposed to use `toSimpleRuleSchedule` only in the UI and only in a read-only fashion?

Reply:
Rule schedule's form is stored non-normalized. There is always a chance prebuilt rule will be marked as customized after saving a form without changes. Rule run interval can set in any units as well as `from` date math. Look-back conversion should transform it to some units. And this task isn't trivial when `interval` and `from` use different units.  

We can use the following to mitigate this issue

- Store only normalized time durations
- Store seconds forced time durations and normalize in UI
- For `interval` units for look-back

--- 

Let's discuss the ideas over Zoom.